index.js

```diff
@@ -188,15 +208,15 @@ FacebookInsightStream.prototype._collect = function ( metrics, item, buffer, eve
 
     // for the audience API, we just use one metric ['app_event']
     // with a few events
-    var _metric = metrics.shift() || options.metrics[ 0 ];
+    var _metric = metrics[ metrics.length -1 ] || options.metrics[ 0 ];
     var model = { id: item.id, metric: _metric }
 
     var _ev;
     var _agg;
     if ( hasEvents ) {
         // extend the query model with event name
         // and aggregation type
-        _ev = events.shift();
+        _ev = events[ events.length - 1 ];
```

Comment:
@alonweissfeld I am not sure its need to be documented as its not unusual, most of our data sources not shifting the resources but removing them after a succeed collect, Also when new developer will look at this comment without knowing we used to shift the resources, he wont understand what this comment is about.